### PR TITLE
Trainer auto wait pserver ports

### PR DESCRIPTION
**Pull request by typhoonzero**

### CODE REVIEW

File: python/paddle/fluid/transpiler/details/checkport.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    port readiness.
+
+    Args:
+        endpoints (list): endpoints string list, like:
+                         ["127.0.0.1:8080", "127.0.0.1:8081"]
+
+    Examples:
+        .. code-block:: python
+
+           check_port_open(["127.0.0.1:8080", "127.0.0.1:8081"])
```
Comment (gongweibao) [inline python/paddle/fluid/transpiler/details/checkport.py]: wait_server_ready

**Review by gongweibao — CHANGES_REQUESTED** [review]

**Comment by gongweibao** [comment]
1. Do we need to clean these? for example:
https://github.com/PaddlePaddle/Paddle/blob/develop/benchmark/fluid/fluid_benchmark.py#L166
https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/fluid/operators/distributed/rpc_server.cc#L69

**Comment by gongweibao** [comment]
是不是用一个OP来实现这个等待更好一点？

**Comment by typhoonzero** [comment]
@gongweibao 分布式都改到pass里之后可能会需要用op

File: python/paddle/fluid/transpiler/details/checkport.py (hunk 1 above)
Comment (typhoonzero) [reply python/paddle/fluid/transpiler/details/checkport.py]: Done

File: python/paddle/fluid/transpiler/distribute_transpiler.py [hunk 2]
```diff
@@ -381,7 +381,7 @@ def transpile(self,
                                                         pserver_endpoints)
             self._split_table_grad_and_add_send_vars(program, pserver_endpoints)
 
-    def get_trainer_program(self):
+    def get_trainer_program(self, wait_port=True):
```
Comment (gongweibao) [inline python/paddle/fluid/transpiler/distribute_transpiler.py]: Maybe we can add  a parameter `dry_run` to control wait or not.

**Review by gongweibao — APPROVED** [review]
LGTM

**Review by qingqing01 — APPROVED** [review]
Approved for API.

**Review by reyoung — APPROVED** [review]

**Pull request merged**